### PR TITLE
Docs: Update transform docs

### PR DESCRIPTION
**Pull request by nmarrs**
Update transform docs for enum convert field support. Also add docs for `Other` convert field type option (see [this PR for context on this option](https://github.com/grafana/grafana/pull/42624))
Screenshot 2023-11-16 at 12 31 36 PM

Related to https://github.com/grafana/grafana/pull/76410

### CODE REVIEW

**Review by imatwawana — COMMENTED** [review]
This LGTM, however, I'm pretty sure you should be updating the following file public/app/features/transformers/docs/content.ts (it's noted in the front matter of this file). @baldm0mma, yes?

**Comment by baldm0mma** [comment]
> This LGTM, however, I'm pretty sure you should be updating the following file public/app/features/transformers/docs/content.ts (it's noted in the front matter of this file). @baldm0mma, yes?

That's correct, thanks @imatwawana! @nmarrs the markdown is now being built from the content in `public/app/features/transformers/docs/content.ts` - we did this so that the documentation can now be available in the app itself for referencing. Please update the documentation in `content.ts`, then run the script mentioned at the top of that file to rebuild the markdown in the `index.md`. Then we'll merge both files.

File: docs/sources/panels-visualizations/query-transform-data/transform-data/index.md [hunk 1]
```diff
@@ -272,6 +272,9 @@ Use this transformation to change the field type of the specified field.
   - **Time -** attempts to parse the values as time
     - Will show an option to specify a DateFormat as input by a string like yyyy-mm-dd or DD MM YYYY hh:mm:ss
   - **Boolean -** will make the values booleans
+  - **Enum -** will make the values enums
+    - Will show a table to manage the enums
+  - **Other -** attempts to parse the values as JSON
```
Comment (baldm0mma) [inline docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:277]:
```suggestion
  - **Enum** - will make the values enums
    - Will show a table to manage the enums
  - **Other** - attempts to parse the values as JSON
```
Just made a quick change to the new syntax, and again, this should be updated in the `content.ts` file, the script mentioned at the top of the file run, then the markdown should reflect the change.

**Comment by nmarrs** [comment]
Ah was not aware that some docs are controlled by TS files 😬 thanks for the heads up! I've updated the PR accordingly and also fixed the list item formatting for the entire file :)

**Review by imatwawana — APPROVED** [review]
LGTM!

**Pull request merged**